R/capacity/health-inequalities/england/quality/cqc-surveys.R

```diff
@@ -0,0 +1,166 @@
+outpatient_ae <- survey_data_download("https://www.cqc.org.uk/sites/default/files/20210915_uec20_type1_benchmark-data.ods", 47, "UEC20_Trust_Scores", "q", "ae")
+outpatient_minor_inj <- survey_data_download("https://www.cqc.org.uk/sites/default/files/20210915_uec20_type3_benchmark-data.ods", 39, "UEC20_Trust_Scores", "t", "min")
+
+# Load in open trusts table created in trust_types.R
+open_trusts <- read_rds("data/open_trust_types.rds")
+
+# Check if any trusts in the surveys not in open_trusts dataset in case any changed trust codes
+# Surveys at different years so may have different stages of trust changes
+inpatient_survey |>
+  anti_join(open_trusts)
+
+mental_health_survey |>
+  anti_join(open_trusts)
+
+outpatient_ae |>
+  anti_join(open_trusts)
+
+outpatient_minor_inj |>
+  anti_join(open_trusts)
+# No updated needed
+
+trust_changes <- read_rds("data/trust_changes.rds")
+
+# Update trust codes using updating_trusts_for_survey_data() function
+inpatient_survey_updated <- updating_trusts_for_survey_data(inpatient_survey, num_respon_inp, meaninp, open_trusts, trust_changes)
+mental_health_survey_updated <- updating_trusts_for_survey_data(mental_health_survey, num_respon_mh, meanmh, open_trusts, trust_changes)
+outpatient_ae_updated <- updating_trusts_for_survey_data(outpatient_ae, num_respon_ae, meanae, open_trusts, trust_changes)
+
+outpatient_minor_inj_updated <- outpatient_minor_inj |>
+  select(trust_code, num_respon = num_respon_min, mean = meanmin)
+
+# Join data survey data to open trust data --------
+combined_survey_data <- open_trusts |>
+  left_join(inpatient_survey_updated, by = "trust_code") |>
+  left_join(mental_health_survey_updated, by = "trust_code") |>
+  left_join(outpatient_ae_updated, by = "trust_code") |>
+  left_join(outpatient_minor_inj_updated, by = "trust_code")
+
+# Not every trust will provide all the service types (e.g. a&e service, mental health service etc) so won't have surveys for all (see next section for check)
+combined_survey_data |>
+  select(trust_code, primary_category, starts_with("num_respon")) |>
+  group_by(primary_category) |>
+  summarise(across(where(is.numeric), ~ sum(!is.na(.x))), count = n())
+# Ambulance services primary providers haven't completed any of the surveys since they don't provide any of the services in the 4 surveys currently including.
+# May be similar for others
+
+# Combining the survey scores ---------
+
+# Have a think/research if way to combine the variability (i.e the upper and lower confidence intervals) for the scores
+response_columns <- str_subset(colnames(combined_survey_data), "num_respon")
+
+# Come up with a better/more reproducible way of writing this query
+avg_survey_scores <- combined_survey_data |>
+  rowwise() %>%
+  mutate(total_responders = sum(c(num_respon.x, num_respon.y, num_respon.x.x, num_respon.y.y), na.rm = T)) |>
```

Comment:
```suggestion
  mutate(
    total_responders = sum(
      c_across(starts_with("num_respon.")) ,
      na.rm = T
    )
  ) |>
```
Better to use tidy select semantics in `c_across` with `rowwise()` for better legibility: https://dplyr.tidyverse.org/reference/c_across.html